### PR TITLE
Enable auto-publishing of module to PyPI.

### PR DESCRIPTION
**Pull request by mithro**
After merging, the module will be auto-published to PyPI.

It will also be checked that it can be installed via;
 - Checkout out the git repository.
 - Directly from GitHub (through `requirements.txt` or `pip` directly).
 - Source and wheel packages.

It also checks the package works on Linux, Mac and Windows.

### CODE REVIEW

File: .github/workflows-src/install-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Use of this source code is governed by a ISC-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
+# SPDX-License-Identifier: ISC
+
+name: Test after installing from...
+
+on:
+  push:
```
Comment (kgugala) [inline .github/workflows-src/install-and-test.yml:12]: shouldn't we limit CI on to run on pushes to `master` only?

File: .github/workflows-src/build-and-upload-for-pypi-pure.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# Use of this source code is governed by a ISC-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
+# SPDX-License-Identifier: ISC
+
+name: "Build (pure python) packages and publish on PyPI"
+
+on:
+  push:
```
Comment (kgugala) [inline .github/workflows-src/build-and-upload-for-pypi-pure.yml:12]: shouldn't we limit CI to run on pushes to `master` only?

**Review by kgugala — CHANGES_REQUESTED** [review]
LGTM in general, some small comments around CI config

File: .github/workflows-src/build-and-upload-for-pypi-pure.yml (hunk 2 above)
Comment (mithro) [reply .github/workflows-src/build-and-upload-for-pypi-pure.yml:12]: It only pushes to PyPI on `master`, the other ones it just builds the files and uploads it as artifacts.

File: .github/workflows-src/install-and-test.yml (hunk 1 above)
Comment (mithro) [reply .github/workflows-src/install-and-test.yml:12]: We most certainly want this to run on any branch created.

**Pull request merged**